packages/flutter/test/material/flexible_space_bar_test.dart

```diff
@@ -117,6 +117,49 @@ void main() {
     expect(clipRect.size.height, minExtent);
   });
 
+  // This is a regression test for https://github.com/flutter/flutter/issues/14227
+  testWidgets('FlexibleSpaceBar sets width constraints for the title', (WidgetTester tester) async {
+    const double height = 300.0;
+    double width;
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Builder(
+            builder: (BuildContext context) {
+              width = MediaQuery.of(context).size.width;
+              return CustomScrollView(
+                slivers: <Widget>[
+                  SliverAppBar(
+                    expandedHeight: height,
+                    pinned: true,
+                    stretch: true,
+                    flexibleSpace: FlexibleSpaceBar(
+                      titlePadding: EdgeInsets.zero,
+                      title: Text(
+                        'X' * 2000,
+                        maxLines: 1,
+                        overflow: TextOverflow.ellipsis,
+                      ),
+                      centerTitle: false,
+                    ),
+                  ),
+                ],
+              );
+            }
+          ),
+        ),
+      ),
+    );
+
+    // The title is scaled and transformed to be 1.5 times bigger, when the
+    // FlexibleSpaceBar is fully expanded, thus we expect the width to be
+    // 1.5 times smaller than the full width.
```

Comment:
Please add a sentence or two that explains where the -30 and -10 adjustments (top,bottom) come from.

Reply:
Done!